lightning/src/blinded_path/payment.rs

```diff
@@ -160,3 +233,162 @@ impl_writeable_msg!(PaymentConstraints, {
+			payment_constraints: PaymentConstraints {
+				max_cltv_expiry: 0,
+				htlc_minimum_msat: 1,
+			},
+		};
+		let blinded_payinfo = super::compute_payinfo(&[], &recv_tlvs, 4242).unwrap();
+		assert_eq!(blinded_payinfo.fee_base_msat, 0);
+		assert_eq!(blinded_payinfo.fee_proportional_millionths, 0);
+		assert_eq!(blinded_payinfo.cltv_expiry_delta, 0);
+		assert_eq!(blinded_payinfo.htlc_minimum_msat, 1);
+		assert_eq!(blinded_payinfo.htlc_maximum_msat, 4242);
+	}
+
+	#[test]
+	fn simple_aggregated_htlc_min() {
+		// If no hops charge fees, the htlc_minimum_msat should just be the maximum htlc_minimum_msat
+		// along the path.
+		let dummy_pk = PublicKey::from_slice(&[2; 33]).unwrap();
+		let intermediate_nodes = vec![(dummy_pk, ForwardTlvs {
+			short_channel_id: 0,
+			payment_relay: PaymentRelay {
+				cltv_expiry_delta: 0,
+				fee_proportional_millionths: 0,
+				fee_base_msat: 0,
+			},
+			payment_constraints: PaymentConstraints {
+				max_cltv_expiry: 0,
+				htlc_minimum_msat: 1,
+			},
+			features: BlindedHopFeatures::empty(),
+		}), (dummy_pk, ForwardTlvs {
+			short_channel_id: 0,
+			payment_relay: PaymentRelay {
+				cltv_expiry_delta: 0,
+				fee_proportional_millionths: 0,
+				fee_base_msat: 0,
+			},
+			payment_constraints: PaymentConstraints {
+				max_cltv_expiry: 0,
+				htlc_minimum_msat: 2_000,
+			},
+			features: BlindedHopFeatures::empty(),
+		})];
+		let recv_tlvs = ReceiveTlvs {
+			payment_secret: PaymentSecret([0; 32]),
+			payment_constraints: PaymentConstraints {
+				max_cltv_expiry: 0,
+				htlc_minimum_msat: 3,
+			},
+		};
+		let htlc_maximum_msat = 100_000;
+		let blinded_payinfo = super::compute_payinfo(&intermediate_nodes[..], &recv_tlvs, htlc_maximum_msat).unwrap();
+		assert_eq!(blinded_payinfo.htlc_minimum_msat, 2_000);
+	}
+
+	#[test]
+	fn aggregated_htlc_min() {
+		// Create a path with varying fees and htlc_mins, and make sure htlc_minimum_msat ends up as the
+		// max (htlc_min - following_fees) along the path.
+		let dummy_pk = PublicKey::from_slice(&[2; 33]).unwrap();
+		let intermediate_nodes = vec![(dummy_pk, ForwardTlvs {
+			short_channel_id: 0,
+			payment_relay: PaymentRelay {
+				cltv_expiry_delta: 0,
+				fee_proportional_millionths: 500,
+				fee_base_msat: 1_000,
+			},
+			payment_constraints: PaymentConstraints {
+				max_cltv_expiry: 0,
+				htlc_minimum_msat: 5_000,
+			},
+			features: BlindedHopFeatures::empty(),
+		}), (dummy_pk, ForwardTlvs {
+			short_channel_id: 0,
+			payment_relay: PaymentRelay {
+				cltv_expiry_delta: 0,
+				fee_proportional_millionths: 500,
+				fee_base_msat: 200,
+			},
+			payment_constraints: PaymentConstraints {
+				max_cltv_expiry: 0,
+				htlc_minimum_msat: 2_000,
+			},
+			features: BlindedHopFeatures::empty(),
+		})];
+		let recv_tlvs = ReceiveTlvs {
+			payment_secret: PaymentSecret([0; 32]),
+			payment_constraints: PaymentConstraints {
+				max_cltv_expiry: 0,
+				htlc_minimum_msat: 1,
+			},
+		};
+		let htlc_minimum_msat = 4798;
```

Comment:
So is this the min amount (excluding fees) someone could send to this path, i.e. by the time an HTLC gets to the introduction node, as long as this node receives an HTLC with amount >= `BlindedPayInfo::htlc_minimum_msat`, it'll be enough to pay the fees (..and meet the following hops' `htlc_minimum_msat` thresholds?) for all the following hops? I saw [this comment](https://github.com/lightningdevkit/rust-lightning/pull/2514#discussion_r1310411839) but don't fully understand what it means